events/app.py

```diff
@@ -0,0 +1,150 @@
+import os
+import logging
+import datetime
+import pymongo
+
+from collections import namedtuple
+from flask import Flask, request, Response
+
+app = Flask(__name__)
+
+
+def connect_to_mongodb():
+    """Connects to MongoDB Atlas database.
+
+    Returns events collection if connection is successful, and None otherwise.
+    """
+    mongodb_uri = os.environ.get("MONGODB_URI")
+    if mongodb_uri is None:
+        print("Alert: not able to find MONGODB_URI environmental variable, "
+              "no connection to MongoDB instance")
+        return None  # not able to find db config var
+    return pymongo.MongoClient(mongodb_uri).eventsDB.all_events
+
+
+EVENTS_COLL = connect_to_mongodb()  # None if can't connect
+
+
+@app.route('/v1/add', methods=['POST'])
+def add_event():
+    # TODO(cmei4444): Get event information from request data
+    info = request.form['info']
+    # TODO(cmei4444): Verify that user has event editing access
+    current_time = datetime.datetime.now()
+    info = build_event_info(info, current_time)
+    try:
+        event = Event(info)
+    except ValueError:      # missing or extra event attributes
+        return Response(
+            status=400,
+            response="Event info was entered incorrectly.",
+        )
+    if EVENTS_COLL is not None:
```

Comment:
If you have EVENTS_COLL be a thing that throws, you can restructure this to:

```python
try:
  event.add_to_db(EVENTS_COL)
  return Response(status=201)
except Exception as e:
  print("Oh no, an exception happened")
  traceback.print_exc()
  return Response(status=500, response="Oops")

Reply:
(The reason for doing this would be to also handle exceptions from e.g. database writes.)

Reply:
Done in https://github.com/knative-portability/large-events/pull/4/commits/ccab3b6f737717ba360d5ef9fb156313f751ead1